### PR TITLE
feat: add an option to `cabal2nix` hook for specifying output filename.

### PR DESCRIPTION
**Pull request by rsrohitsingh682**
At present after enabling `cabal2nix` hook, it generates a file named `default.nix` which is hardcoded. It is better if user has an option to specify the filename instead of using hardcoded `default.nix`.

This PR adds an option to `cabal2nix` hook to specify output filename generated by running `cabal2nix`.

Example: 

```
cabal2nix = {
   enable = true;
   settings.output_filename = "cabal.nix";
};
```

### CODE REVIEW

**Comment by rsrohitsingh682** [comment]
@sandydoo @NorFairKing Any thoughts on this PR ?

**Comment by NorfairKing** [comment]
I don't see a use case so think this needlessly complicating things but I also don't really care :shrug:

**Comment by rsrohitsingh682** [comment]
> I don't see a use case so think this needlessly complicating things but I also don't really care 🤷

It's required for me as I don't want pre-commit hook to generate a `default.nix`.  Anyways since the default is `default.nix` so the default behaviour still remains the same.

**Comment by rsrohitsingh682** [comment]
@sandydoo Could you please help with the review ?

File: modules/hooks.nix [hunk 1]
```diff
@@ -208,6 +208,20 @@ in
           };
         };
       };
+      cabal2nix = mkOption {
+        description = "cabal2nix hook";
+        type = types.submodule {
+          imports = [ hookModule ];
+          options.settings = {
+            output_filename =
```
Comment (sandydoo) [inline modules/hooks.nix:216]: Lets stick with camelCase for the settings, if possible.

**Review by sandydoo — CHANGES_REQUESTED** [review]
Otherwise, LGTM

**Comment by sandydoo** [comment]
The CI failure is unrelated. This branch needs a rebase on master.

**Review by sandydoo — APPROVED** [review]

**Pull request merged**